lib/cache.rb

```diff
@@ -74,17 +74,16 @@ def delete(name)
 
   def fetch(name, expires_in: nil, force: nil, &blk)
     if !block_given?
-      if force
-        raise ArgumentError,
-              "Missing block: Calling `Cache#fetch` with `force: true` requires a block."
-      end
+      raise ArgumentError, "Calling `Cache#fetch` with `force: true` requires a block." if force
       return read(name)
     end
 
     key = normalize_key(name)
-    raw = redis.get(key) if !force
-    entry = read_entry(key) if raw
-    return entry if raw && !(entry == :__corrupt_cache__)
+
+    if !force
+      entry = read_entry(key)
+      return entry if entry != nil && entry != :__corrupt_cache__
```

Comment:
Does this mean we can no longer cache `nil` values? Sometimes that's deliberate.

Reply:
To keep the old behaviour without the race condition, I guess it should be something along the lines of


```ruby
if !force
  raw = redis.get(key)
  if raw
    entry = decode_entry(raw)
    return entry if entry != :__corrupt_cache__
  end
end
```